content/docs/1.7.0/advanced-resources/deploy/revision_counter.md

```diff
@@ -3,15 +3,15 @@ title: Revision Counter
 weight: 7
 ---
 
-The revision counter is a strong mechanism that Longhorn uses to track each replica's updates.
+The revision counter is a mechanism that Longhorn uses to track each replica's updates.
 
 During replica creation, Longhorn will create a 'revision.counter' file with its initial counter set to 0. And for every write to the replica, the counter in 'revision.counter' file will be increased by 1.
 
-The Longhorn Engine will use these counters to make sure all replicas are consistent during start time. These counters are also used during salvage recovery to decide which replica has the latest update.
+The Longhorn Engine will use these counters as a heuristic to make a best effort to ensure all replicas are consistent during startup. Note that because the write IOs in Longhorn are parallel, data consistency guarantee via revision counter mechanism is best-effort only. These counters are also used during salvage recovery to decide which replica has the latest update.
 
-Disable Revision Counter is an option in which every write on replicas is not tracked. When this setting is used, performance is improved, but the strong tracking for each replica is lost. This option can be helpful if you prefer higher performance and have a stable network infrastructure (e.g. an internal network) with enough CPU resources. When the Longhorn Engine starts, it will skip checking the revision counter for all replicas, but auto-salvage will still be supported through the replica's head file stat. For details on how auto-salvage works without the revision counter, refer to [this section.](#auto-salvage-support-with-revision-counter-disabled)
+Disable Revision Counter is an option in which every write on replicas is not tracked. When this setting is used, performance is improved. This option can be helpful if you prefer higher performance and have a stable network infrastructure (e.g. an internal network) with enough CPU resources. When the Longhorn Engine starts, it will skip checking the revision counter for all replicas, but auto-salvage will still be supported through the replica's head file stat. For details on how auto-salvage works without the revision counter, refer to [this section.](#auto-salvage-support-with-revision-counter-disabled)
```

Comment:
```suggestion
The *Disable Revision Counter* option prevents Longhorn from tracking every write operation on replicas. This option can be useful if you have a stable network infrastructure with sufficient CPU resources, and want to maximize the performance of those resources.

When the revision counter is disabled, the Longhorn Engine skips checking the revision counter for all replicas at startup. However, auto-salvage still functions because Longhorn can use certain file properties to identify the replica to be used for recovery. For more information about how auto-salvage functions without the revision counter, see [Auto-Salvage Support with Revision Counter Disabled](#auto-salvage-support-with-revision-counter-disabled).
```

Reply:
Thank you. I incorporated your feedback